src/eos-repo-server.c

```diff
@@ -403,20 +403,29 @@ eos_filez_read_data_disconnect_and_clear_msg (EosFilezReadData *read_data)
 
 static void
 update_pending_requests (EosUpdaterRepoServer *server,
-                         guint new_value)
+                         gint                  delta)
 {
-  if (server->pending_requests == new_value)
+  GObject *obj = G_OBJECT (server);
+
+  g_assert (delta >= 0 || server->pending_requests >= (guint) -delta);
+  g_assert (delta <= 0 || server->pending_requests <= G_MAXUINT - delta);
+
+  if (delta == 0)
```

Comment:
Because the number of requests hasn’t changed, so there is no new ‘last request’ to change the time for.